index/scorch/segment/segment.go

```diff
@@ -75,6 +75,11 @@ type PostingsIterator interface {
 	Next() (Posting, error)
 
 	Size() int
+
+	// Advance will return the respective posting by loading
+	// the chunk in which the doc detail resides and calling Next() until
+	// it finds the given doc details.
```

Comment:
This doc-comment for Advance feels a bit too much tied to the proposed initial implementation, when it mentions "calling Next()" in a loop.

For example, I can imagine a future segment implementation (zap2?) that uses an idea like your docValues approach of having an index... so that it might jump more directly to the right place?

That is, I'm thinking the method doc comment should more about what-not-how?